Formula/blackbox.rb

```diff
@@ -3,18 +3,39 @@ class Blackbox < Formula
   homepage "https://github.com/StackExchange/blackbox"
   url "https://github.com/StackExchange/blackbox/archive/v1.20160122.tar.gz"
   sha256 "ac5de1d74fdbe88604b34949f3949e53cb72e55e148e46b8c2be98806c888a10"
+  revision 1
 
   bottle :unneeded
 
-  depends_on :gpg
+  depends_on :gpg => :run
 
   def install
+    # Prefer GPG2 by default.
+    inreplace "bin/_blackbox_common.sh", ":=gpg", ":=gpg2"
+
     libexec.install Dir["bin/*"]
     bin.write_exec_script Dir[libexec/"*"].select { |f| File.executable? f }
   end
 
   test do
+    # Note that test do is always in temporary path, with HOME captured,
+    # so this doesn't interfere with a user's pre-existing keychain.
+    (testpath/"batchgpg").write <<-EOS.undent
+    Key-Type: RSA
```

Comment:
It's planned:

> Eventually the batch generation of a gpg key for proper functionality testing will be split into a Utils method to call, rather than having to write the whole blob into every formula each time, so that is a temporary measure rather than a permanent thing.